File: 002.js
'use strict';

var fibonacci = function(max) {
  var fibArray = [1, 2];

  for (var i = 1; i<= max; i++) {
      fibArray[i+1]=(fibArray[i] + fibArray[i-1]);
  }
  return fibArray;
};


var sumEvens = function(max) {
    var fibArray = fibonacci(max);
    var evenArray = [];
    var sum;

    for (var i = 0; i < fibArray.length; i++) {
        if (fibArray[i] % 2 === 0) {
            evenArray.push(fibArray[i]);
        }
    }
    sum = evenArray.reduce(function(total, num){
        return total + num }
  , 0);
    return sum;
};

// gives sum of even numbers whose values don't
// exceed 4 million
sumEvens(30);

File: 002.rb
def fibonacci(max)
    fib_array = [1, 2]
    (1..max).each do |n|
        fib_array.push(fib_array[n] + fib_array[n-1])
    end
    return fib_array
end

def sum_evens(max)
    fib_array = fibonacci(max)
    fib_array.select(&:even?).reduce(:+)
end

sum_evens(30)
